Client/HttpCookieExtension.swift

```diff
@@ -0,0 +1,84 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+import Foundation
+import WebKit
+import Shared
+
+private let log = Logger.browserLogger
+
+public extension HTTPCookie {
+    
+    typealias Success = Bool
+    class func saveToDisk(_ filename: String = "CookiesData.json", completion: ((Success) -> Void)? = nil) {
+        let cookieStore = WKWebsiteDataStore.default().httpCookieStore
+        
+        /* For reason unkown the callback to getAllCookies is not called, when the save is done from Settings.
+         A possibility is https://bugs.webkit.org/show_bug.cgi?id=188242
+         Even with the issue being fixed it still sometimes doesn't work.
+         The network process is in suspened maybe?
+         
+         And for some reason fetch cookie records preemptively guarantees this works.
+         Best guess is that fetching records brings network process to active.
+         
+         Same applies to setting cookies back below.
+         */
+        WKWebsiteDataStore.default().fetchDataRecords(ofTypes: [WKWebsiteDataTypeCookies]) { (_) in}
+        cookieStore.getAllCookies { (cookies) in
+            let baseDir = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
+            do {
+                let data = try NSKeyedArchiver.archivedData(withRootObject: cookies, requiringSecureCoding: false)
+                try data.write(to: URL(fileURLWithPath: "\(baseDir)/\(filename)"))
+            } catch {
+                log.error("Failed to write cookies to disk with error: \(error)")
+                completion?(false)
+                return
+            }
+            completion?(true)
+        }
+    }
+    
+    class func loadFromDisk(_ filename: String = "CookiesData.json", completion: ((Success) -> Void)? = nil) {
+        let baseDir = NSSearchPathForDirectoriesInDomains(.documentDirectory, .userDomainMask, true)[0]
+        do {
+            let data = try Data(contentsOf: URL(fileURLWithPath: "\(baseDir)/\(filename)"), options: Data.ReadingOptions.alwaysMapped)
```

Comment:
see `saveToDisk` comments, use appending component instead of hardcoding file path

Reply:
Cant append Component to String and cant use URL here until this point.